### PR TITLE
Documentation for control plane high availability

### PR DESCRIPTION
**Pull request by mviitane**
**What this PR Includes**
- Major update on high availability documentation
- Other minor docs improvements

Signed-off-by: mviitanen <mviitanen@mirantis.com>

### CODE REVIEW

File: docs/high-availability.md [hunk 1]
```diff
@@ -1,22 +1,86 @@
...
+
+### Configuration using k0s.yaml (for each controller)
+
+Note to update your load balancer's public ip address into two places.
 
-- 6443
-- 8132
-- 8133
-- 9443
+```sh
```
Comment (kke) [inline docs/high-availability.md:64]:
```suggestion
```yaml
```

File: docs/high-availability.md [hunk 2]
```diff
@@ -1,22 +1,86 @@
...
-- `externalAddress`
+```sh
+  "k0s":
+    "config":
+      "spec":
+        "api":
+          "externalAddress": "<load balancer public ip address>"
+          "sans":
+          - "<load balancer public ip address>"
+```
```
Comment (kke) [inline docs/high-availability.md:84]: maybe replace ugly quoted yaml with pretty yaml :) (and use ```yaml)

Comment (mviitane) [reply docs/high-availability.md:84]: Thanks! Good comments. To be fixed.

File: docs/high-availability.md (hunk 1 above)
Comment (mviitane) [reply docs/high-availability.md]: fixed

File: docs/high-availability.md [hunk 3]
```diff
@@ -1,22 +1,86 @@
...
+
+![k0s high availability](img/k0s_high_availability.png)
+
+## Network considerations
+
+You should plan to allocate the control plane nodes into different zones. This will avoid failures in case one zone fails.
+
+For etcd high availability it's recommended to configure 3 or 5 controller nodes.
+
+- With 3 nodes it's possible to lose one node and still have the quorum with 2 working nodes.
```
Comment (trawler) [inline docs/high-availability.md:13]: maybe we can add a link to the official etcd faqs about this subject? https://etcd.io/docs/v3.3/faq/#why-an-odd-number-of-cluster-members

**Review by trawler — APPROVED** [review]

**Pull request merged**